app/assets/src/components/views/discovery/DiscoverySidebar.jsx

```diff
@@ -30,138 +21,100 @@ export default class DiscoverySidebar extends React.Component {
         avgNonHostReads: ""
       },
       metadata: {
-        host: {},
-        tissue: {},
-        createdAt: {},
-        location: {}
+        host: [],
+        tissue: [],
+        time: [],
+        location: []
       },
       expandedMetadataGroups: new Set()
     };
   }
 
   static getDerivedStateFromProps(newProps, prevState) {
-    const { currentTab, projects, loading } = newProps;
+    const {
+      currentTab,
+      loading,
+      projectDimensions,
+      projects,
+      sampleDimensions
+    } = newProps;
     if (loading) return prevState;
 
-    if (currentTab === "samples") {
-      const samples = DiscoverySidebar.selectSampleData(newProps.samples);
-      return {
-        stats: {
-          numSamples: samples.length,
-          numProjects: uniqBy("project_id", samples).length,
-          avgTotalReads: DiscoverySidebar.meanByAndFormat(
-            samples,
-            "totalReads"
-          ),
-          avgNonHostReads: DiscoverySidebar.meanByAndFormat(
-            samples,
-            "nonHostReads"
-          )
-        },
-        metadata: {
-          host: countBy("hostGenome", samples),
-          tissue: countBy("sampleTissue", samples),
-          createdAt: countBy("createdAt", samples),
-          location: countBy("collectionLocation", samples)
-        }
-      };
-    } else if (currentTab === "projects") {
-      const hosts = flatten(map("hosts", projects));
-      const tissues = flatten(map("tissues", projects));
-      const locations = flatten(map("locations", projects));
+    const dimensions =
+      currentTab === "projects" ? projectDimensions : sampleDimensions;
 
-      const createdAts = map(
-        project => DiscoverySidebar.formatDate(project.created_at),
-        projects
-      );
-
-      return {
-        stats: {
-          numSamples: sumBy("number_of_samples", projects),
-          numProjects: projects.length
-        },
-        metadata: {
-          host: countBy(null, hosts),
-          tissue: countBy(null, tissues),
-          createdAt: countBy(null, createdAts),
-          location: countBy(null, locations)
-        }
-      };
-    } else {
-      // eslint-disable-next-line no-console
-      console.error("Not supported: " + currentTab);
-      return prevState;
-    }
+    return {
+      stats: {
+        numSamples: sumBy("number_of_samples", projects),
+        numProjects: projects.length,
+        avgTotalReads: DiscoverySidebar.meanByAndFormat(
+          newProps.samples,
+          "totalReads"
+        ),
+        avgNonHostReads: DiscoverySidebar.meanByAndFormat(
+          newProps.samples,
+          "nonHostReads.value"
+        )
+      },
+      metadata: {
+        host: (find({ dimension: "host" }, dimensions) || {}).values || [],
```

Comment:
could make a util function out of this extractor

Reply:
Done.